third_party/amd/backend/compiler.py

```diff
@@ -71,6 +72,36 @@ def hash(self):
         return hashlib.sha256(key.encode("utf-8")).hexdigest()
 
 
+class HIPAttrsDescriptor(AttrsDescriptor):
+    __slots__ = ("pointer_range_32")
+
+    @staticmethod
+    def is_within2gb(arg):
+        if hasattr(arg, "ptr_range"):
+            return arg.ptr_range() < 2**31 - 1
```

Comment:
`<=` or `<`?

Reply:
Actually forgot that we are using the 2*31 case as the oob case.. Could you add a comment here to explain so it's easier for others to understand.

Reply:
I think <= was the right thing